contracts/test.sol

```diff
@@ -143,25 +134,61 @@ contract TestUtils is DSTest {
 
 contract InvariantTest {
 
-    address[] private _excludedContracts;
-    address[] private _targetContracts;
-
-    function addTargetContract(address newTargetContract_) internal {
-        _targetContracts.push(newTargetContract_);
+    struct FuzzSelector {
+        address addr;
+        bytes4[] selectors;
     }
 
-    function targetContracts() public view returns (address[] memory targetContracts_) {
-        require(_targetContracts.length != uint256(0), "NO_TARGET_CONTRACTS");
-        return _targetContracts;
-    }
+    address[] private _excludedContracts;
+    address[] private _excludedSenders;
+    address[] private _targetedContracts;
+    address[] private _targetedSenders;
+
+    FuzzSelector[] internal _targetedSelectors;
 
     function excludeContract(address newExcludedContract_) internal {
         _excludedContracts.push(newExcludedContract_);
     }
 
     function excludeContracts() public view returns (address[] memory excludedContracts_) {
-        require(_excludedContracts.length != uint256(0), "NO_TARGET_CONTRACTS");
-        return _excludedContracts;
+        require(_excludedContracts.length != uint256(0), "NO_EXCLUDED_CONTRACTS");
+        excludedContracts_ = _excludedContracts;
+    }
+
+    function excludeSender(address newExcludedSender_) internal {
+        _excludedSenders.push(newExcludedSender_);
+    }
+
+    function excludeSenders() public view returns (address[] memory excludedSenders_) {
+        require(_excludedSenders.length != uint256(0), "NO_EXCLUDED_SENDERS");
+        excludedSenders_ = _excludedSenders;
+    }
+
+    function targetContract(address newTargetedContract_) internal {
```

Comment:
`nitpick`: Could we pass in arrays instead? Usually we are targeting multiple contracts. (same in other functions that use `push`)

Reply:
While we could, then you'd need to build a dynamic array and pass it, which just seems like extra steps and is uglier (especially for larger lists, imo) that several lines of `targetContract`.